spec.markdown

```diff
@@ -348,19 +348,11 @@ only deliver integrity metadata on a [potentially secure origin][].  See
 
 The following algorithm details these restrictions:
 
-1.  Let <var>request</var> be the request that fetched
-    <var>resource</var>.
-2.  If the [response type][] is `opaque`, return `false`.
-3.  If the [mode][fetch-mode] of <var>request</var> is `CORS`,
-    return `true`.
-4.  If the [origin][fetch-origin] of <var>request</var> is
-    <var>resource</var>'s origin, return `true`.
-5.  Return `false`.
-
-Step 3 returns `true` if the fetch was a CORS-enabled request. If the
-fetch failed the CORS checks, it won't be available to us for integrity
-checking because it won't have loaded successfully.
-{:.note}
+1.  Let <var>response</var> be the response that results from
+    fetching the <var>resource</var>.
+2.  If the <var>response</var> [type][response type] is `basic`,
+    `cors` or `default`, return `true`.
+3.  Return `false`.
 
```

Comment:
I think we should add a non-normative note that reads something like this:

The `response types` are defined by the Fetch spec, and refer to the following:
   `basic` is a same-origin response, and thus the requestor has full access to read the content.
   `cors` is a valid response to a CORS-enabled request, and thus again the requestor has full access to read the content.
    `default` is a valid response that is generated by a Service Worker as a response to the request, so it, too, is fully readable by the requestor.